### DISCUSSION TITLE
plugin implementation: close on ESC

### DISCUSSION THREAD

**Discussion by Kolcha** [doublecmd/doublecmd#1623] [General]
I've implemented my own viewer plugin, but when it is used, viewer can't be closed using ESC key, what is very inconvenient.

I thought "close on ESC" is built-in viewer's feature, as it works "seamlessly" for any TC plugins I used on Windows, but as experiments shown it doesn't work for any plugins I tried to build (no only mine) on Linux.

so, my question: what I (as plugin's author) can (and should) do to get this convenient "close on ESC" feature to work? I'm using Qt-based DC (if this is relevant), plugin is implemented in C++.

**Comment by j2969719** [comment]
if you are using qt build, in the hotkey settings in the viewer category, remove the second hotkey (Q key).  there seems to be some problem with lazarus-qt iirc
